Source/objects.h

```diff
@@ -58,28 +63,64 @@ struct ObjectStruct {
 	 * antechamber levers). The coordinates used for this region are based on a 40*40 grid overlaying the central
 	 * 80*80 region of the dungeon.
 	 *
-	 * @param topLeft corner of the map region closest to the origin
-	 * @param bottomRight corner of the map region furthest from the origin
-	 * @param v ID/discriminator for the object type? Needs to be investigated further
+	 * @param topLeftPosition corner of the map region closest to the origin.
+	 * @param bottomRightPosition corner of the map region furthest from the origin.
+	 */
+	constexpr void SetMapRange(Point topLeftPosition, Point bottomRightPosition)
+	{
+		_oVar1 = topLeftPosition.x;
+		_oVar2 = topLeftPosition.y;
+		_oVar3 = bottomRightPosition.x;
+		_oVar4 = bottomRightPosition.y;
+	}
+
+	/**
+	 * @brief Convenience function for SetMapRange(Point, Point).
+	 * @param mapRange A rectangle defining the top left corner and size of the affected region.
+	 */
+	constexpr void SetMapRange(Rectangle mapRange)
+	{
+		SetMapRange(mapRange.position, mapRange.position + Displacement { mapRange.size });
+	}
+
+	/**
+	 * @brief Sets up a generic quest book which will trigger a change in the map when activated.
+	 *
+	 * Books of this type use a generic message (see OperateSChambBook()) compared to the more specific quest books
+	 * initialized by IntializeQuestBook().
+	 *
+	 * @param mapRange The region to be updated when this object is activated.
+	*/
+	constexpr void InitializeBook(Rectangle mapRange)
+	{
+		SetMapRange(mapRange);
+		_oVar6 = _oAnimFrame + 1; // Save the frame number for the open book frame

```

Comment:
I'm pretty sure using `_oVar6` this way is unnecessary, it's used in OperateBookLever and OperateSChambBk to determine if the player has interacted with the object, but this check can also be accomplished by looking at the quest state (done earlier in OperateBookLever) or using this as a bool value. Changing that would require adding logic when saving/loading for compatibility.

Reply:
I do prefer isolating the bad architecture in the save/load routine